src/repository.c

```diff
@@ -638,9 +638,12 @@ static int _git_repository_open_ext_from_env(
 		}
 	}
 
-	error = git_repository_set_namespace(repo, git_buf_cstr(&namespace_buf));
-	if (error < 0)
-		goto error;
+	if (git_buf_len(&namespace_buf)) {
+		error = git_repository_set_namespace(repo, git_buf_cstr(&namespace_buf));
+		if (error < 0)
+			goto error;
+	}
+	error = 0;
```

Comment:
An additional note, which is not your fault but might be improved when we're already touching this. I think the code would be nicer if you moved the call to `git__getenv` and setting the namespace together. So you could just move down getting the environment variable and then set the namespace based on the function's return value.